CommunityEntity.UI.BezierEasing.cs

```diff
@@ -0,0 +1,137 @@
+using UnityEngine;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Facepunch.Extend;
+using System.IO;
+
+#if CLIENT
+
+public partial class CommunityEntity
+{
+
+    public class BezierEasing {
+        // this basically is the same process that Browsers use for their CSS easing, allowing you to define any possible easing value with 4 bezier points
+        // had to port this from a javascript version
+
+        public static Dictionary<string, BezierEasing> cache = new Dictionary<string, BezierEasing>();
```

Comment:
This dictionary should probably be a Dictionary<CustomStruct,BezierEasing> to avoid string allocations every time you access it

Reply:
you're right, thank you for reminding me about string allocations. added the struct: 
![image](https://user-images.githubusercontent.com/33698270/204252709-f56d7aea-29f1-4cc2-9f20-cb24ae5e224b.png)